invenio_administration/assets/semantic-ui/js/invenio_administration/details/ActionModal.js

```diff
@@ -0,0 +1,58 @@
+// This file is part of InvenioAdministration
+// Copyright (C) 2022 CERN.
+//
+// Invenio RDM Records is free software; you can redistribute it and/or modify it
+// under the terms of the MIT License; see LICENSE file for more details.
+
+import PropTypes from "prop-types";
+import React, { Component } from "react";
+import { i18next } from "@translations/invenio_administration/i18next";
+import { Modal, Header } from "semantic-ui-react";
+import { Cancel } from "../actions/Cancel";
+
+class ActionModal extends Component {
+  render() {
+    const {
+      action,
+      trigger,
+      modalId,
+      children,
+      modalOpen,
+      toggleModal,
+      actionTrigger,
+    } = this.props;
+
+    return (
+      <Modal role="dialog" id={modalId} open={modalOpen} trigger={trigger}>
+        <Modal.Header>
+          <Header as="h2">{i18next.t(action)}</Header>
+        </Modal.Header>
+
+        <Modal.Content>
+          <Modal.Description>{children}</Modal.Description>
+        </Modal.Content>
+
+        <Modal.Actions>
+          <Cancel onClick={toggleModal} floated="left" />
+          {actionTrigger}
+        </Modal.Actions>
+      </Modal>
+    );
+  }
+}
+
+export default ActionModal;
+
+ActionModal.propTypes = {
+  action: PropTypes.string.isRequired,
+  children: PropTypes.object,
+  modalId: PropTypes.string.isRequired,
```

Comment:
what do we need the modal id for?

Reply:
For the `aria-controls` of the trigger button. But I see it's not in sync with the aria-attributes. Will update.